### ISSUE TITLE
warmup for BertAdam

### ISSUE THREAD

**Issue by yangdechuan** [huggingface/pytorch-pretrained-BERT#670]
https://github.com/huggingface/pytorch-pretrained-BERT/blob/ee0308f79ded65dac82c53dfb03e9ff7f06aeee4/examples/run_classifier.py#L860

BertAdam() can update learning rate by itself.
 Why update learning rate manually here?

**Comment by thomwolf** [comment]
Because we don't use BertAdam in fp16 mode but the optimizer of NVIDIA's apex library.

**Comment by yangdechuan** [comment]
OK thank you!
